src/test/java/org/littleshoot/proxy/test/SocketClientUtil.java

```diff
@@ -61,10 +61,10 @@ public static String readStringFromSocket(Socket socket) throws IOException {
     public static boolean isSocketReadyToWrite(Socket socket) throws IOException {
         OutputStream out = socket.getOutputStream();
         try {
-            out.write(0);
-            out.flush();
-            out.write(0);
-            out.flush();
+            for(int i = 0; i < 500; ++i) {
```

Comment:
This is just more reliable way of checking if the connection allows writes (trying to write twice is not sufficient).